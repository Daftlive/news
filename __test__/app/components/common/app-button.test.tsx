import {
  render,
  screen,
  fireEvent,
  waitFor,
} from "@testing-library/react-native";
import AppButton from "../../../../app/components/common/app-button";

describe("<AppButton>", () => {
  describe("as default", () => {
    const buttonCallbackFunction = jest.fn(() => null);
    const title = "Test Title";
    const testRenderer = render(
      <AppButton title={title} onPress={buttonCallbackFunction} />
    );

    beforeEach(() => {
      render(<AppButton title={title} onPress={buttonCallbackFunction} />);
    });

    it("Renders successfully", () => {
      expect(testRenderer).toBeDefined();
    });

    it("renders correctly", () => {
      const activeButtonAsJson = testRenderer.toJSON();
      expect(activeButtonAsJson).toMatchSnapshot();
    });

    it("Outputs the correct text", () => {
      expect(screen.getByText(title)).toBeTruthy();
    });

    it("Fires the assigned event on press", async () => {
      fireEvent.press(screen.getByText(title));
      fireEvent.press(screen.getByText(title));
      await waitFor(() =>
        expect(buttonCallbackFunction.mock.calls.length).toBe(2)
      );
    });
  });
});
