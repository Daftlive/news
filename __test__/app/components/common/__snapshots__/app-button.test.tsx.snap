// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppButton> as default renders correctly 1`] = `null`;
